pipreqs/pipreqs.py

```diff
@@ -128,6 +147,8 @@ def get_all_imports(
                     continue
                 else:
                     logging.error("Failed on file: %s" % file_name)
+                    if filter_ext(file_name, [".ipynb"]):
+                        logging.error("Magic command without % might be failed")
```

Comment:
Basically, jupyter notebooks have this feature called _Magic commands_, which give new possible commands to our program using IPython support. Magic commands can be called using the "%" sign, but there's a way to call those commands without using this specific symbol (check for `%automagic` in https://ipython.readthedocs.io/en/stable/interactive/magics.html). This log error basically says that **maybe** the program failed on the file because of that magic command conversion. As far as I know, those magic commands **probably** won't make our program fail, because we are only looking for imports with the AST tree.

Reply:
please add a test to check if magic commands without % would really make pipreqs to fail. If it is not an issue we can drop this if

Reply:
Just created a new file for magic commands inside the jupyter notebook test case (the file is named `magic_commands.ipynb`). As it turns out, tests are running successfully. It seems that magic commands with or without the percentage sign ("_%_") don't affect the ast search. 

With that in mind, i deleted the **if** related to this issue.

Reply:
Oddly enough, just noticed my tests ran successfully when run individually but failed when running with tox. I'll fix this asap

Reply:
Just fixed the issue. As it turns out, tox was not considering the jupyter optional dependency when running it's tests, thus making my local test diverge from tox tests. Just added a line in the `tox.ini` doc to consider the optional dependency when testing. @alan-barzilay